types/plugin.d.ts

```diff
@@ -125,9 +125,17 @@ declare namespace fastifyCookie {
     signed?: boolean;
   }
 
+  enum HookType {
+    onRequest = "onRequest",
+    preHandler = "preHandler",
+    preParsing="preParsing",
+    preValidation="preValidation",
+    preSerialization="preSerialization"
```

Comment:
```suggestion
    preParsing = "preParsing",
    preValidation = "preValidation",
    preSerialization = "preSerialization"
```

Reply:
actually we dont need a enum here. 

type HookType = 'onRequest' | 'preHandler' | 'preParsing' | 'preValidation' | 'preSerialization';